details/operator-sdk/controllers/astraconnector_controller.go

```diff
@@ -196,6 +204,38 @@ func (r *AstraConnectorController) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 }
 
+func (r *AstraConnectorController) deleteNeptuneResources(ctx context.Context, namespace string) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("deleting neptune resources")
+
+	// Get a list of all installed CRDs
+	crdList := &apiextensionsv1.CustomResourceDefinitionList{}
+	if err := r.Client.List(ctx, crdList); err != nil {
+		log.Error(err, "Unable to list CRDs")
+		return
+	}
+
+	for _, crd := range crdList.Items {
+		// Only look for the Neptune ones
+		if strings.Contains(crd.Name, "astra") && !strings.Contains(crd.Name, "astraconnectors.astra.netapp.io") {
```

Comment:
The contains should be more specific "astra.netapp.io", companies like astrazenca could contain astra crd possibly and this would be deleting those crd.

Reply:
Good point